### PR TITLE
HW2 Implement my own Array methods

### PR DESCRIPTION
**Pull request by blooond**
Implementation of some simple Array methods

### CODE REVIEW

**Comment by B0RlS** [comment]
1) You have a little chaos with commits
2) Why did you delete the two readme files?

**Comment by blooond** [comment]
> 1) You have a little chaos with commits
> 
> 2) Why did you delete the two readme files?

1) What's wrong?

2) It was an accident.

File: my_methods.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+module ArrayMethods
```
Comment (B0RlS) [inline my_methods.rb:1]: Could you change the file name please?

File: my_methods.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      return_array = []
+      length.times do |i|
+        return_array << self[i] if yield(self[i])
+      end
+      return_array
+    end
+    
+    def my_empty?
+      true if length.zero?
+      false if length != 0
```
Comment (B0RlS) [inline my_methods.rb:35]: Use ternary conditional operation

File: my_methods.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    def my_reverse!
+      array = Array.new(length)
+      length.times do |i|
+        array[-1 - i] = self[i]
+      end
+      length.times do |i|
+        self[i] = array[i]
+      end
+      self
+    end
```
Comment (B0RlS) [inline my_methods.rb:55]: You don't have this in the requirements

**Review by B0RlS — COMMENTED** [review]
> > 1. You have a little chaos with commits
> > 2. Why did you delete the two readme files?
> 
> 1. What's wrong?
> 2. It was an accident.

- The first two commits are not entirely clear and unnecessary. These commits can be edited, consider using the git revert* command to rewrite the commit history
- I understand, so I ask you to remove the deletion of two files from your PR

File: my_methods.rb (hunk 3 above)
Comment (blooond) [reply my_methods.rb:55]: But Sasha said that if someone wants to do extra tasks you're welcome

File: my_methods.rb (hunk 1 above)
Comment (blooond) [reply my_methods.rb]: Done

File: my_methods.rb (hunk 2 above)
Comment (blooond) [reply my_methods.rb]: Done

File: my_methods.rb (hunk 3 above)
Comment (B0RlS) [reply my_methods.rb]: yes, everything is ok, you can ignore this comment

**Comment by blooond** [comment]
> > > 1. You have a little chaos with commits
> > > 2. Why did you delete the two readme files?
> > 
> > 
> > 
> > 1. What's wrong?
> > 2. It was an accident.
> 
> * The first two commits are not entirely clear and unnecessary. These commits can be edited, consider using the git reset command to rewrite the commit history
> * I understand, so I ask you to remove the deletion of two files from your PR

So I need just to delete my first 2 commits with the help of git reset --soft for example?

**Comment by vladhilko** [comment]
You just need to return back these 2 files somehow, in another case they are will be removed for everyone. If you don't want to fix commit just add them manually 

Screenshot 2021-04-21 at 20 20 57

**Comment by blooond** [comment]
> You just need to return back these 2 files somehow, in another case they are will be removed for everyone. If you don't want to fix commit just add them manually.
> 
> You can also use `git revert` command to remove these 2 commits
> 
> ```ruby
> git revert 855fe3a6db22b0954994660fe0f8bba6c49bebb2
> git revert 50f034693e4a98f778b54ad468723b754f1add42
> ```
> 
> Screenshot 2021-04-21 at 20 20 57

Thanks for help, done

**Review by vladhilko — APPROVED** [review]

**Comment by vladhilko** [comment]
Screenshot 2021-04-21 at 21 42 31

**Comment by vladhilko** [comment]
Closed accidentally 😅

**Comment by B0RlS** [comment]
Could you also change the file name please?
You should give the filename the same as the module name.

**Comment by blooond** [comment]
> Could you also change the file name please?
> You should give the filename the same as the module name.

Done

**Comment by B0RlS** [comment]
> Looks like you forgot to create a new folder with your name
> 
> Screenshot 2021-04-21 at 21 42 31

@blooond

**Comment by blooond** [comment]
> > Looks like you forgot to create a new folder with your name
> > Screenshot 2021-04-21 at 21 42 31
> 
> @blooond

Done

**Pull request merged**